File: v2ray-engine/src/main/java/libv2ray/CoreController.java
package libv2ray;

public class CoreController {
    private final CoreCallbackHandler handler;
    private boolean running = false;

    public CoreController(CoreCallbackHandler handler) {
        this.handler = handler;
    }

    public boolean isRunning() {
        return running;
    }

    public void startLoop(String json, int tunFd) {
        running = true;
        if (handler != null) {
            handler.startup();
        }
    }

    public void stopLoop() {
        running = false;
        if (handler != null) {
            handler.shutdown();
        }
    }

    public long queryStats(String tag, String link) {
        return 0L;
    }

    public long measureDelay(String url) {
        return -1L;
    }
}

File: v2ray-engine/src/main/java/libv2ray/CoreController.kt
package libv2ray

/** In-memory stub; real AAR provides JNI-backed controller. */
class CoreController {
  var isRunning: Boolean = false
    private set

  fun startLoop(config: String, tunFd: Int) {
    isRunning = true
  }

  fun stopLoop() {
    isRunning = false
  }

  fun queryStats(tag: String, link: String): Long = 0L

  fun measureDelay(url: String): Long = -1L
}
